src/utils.js

```diff
@@ -9,8 +10,59 @@ var PLUGIN_NAME = require('../package.json').name;
 
 var urlRegex = /^(https?|webpack(-[^:]+)?):\/\//;
 
+var debug = require('debug-fabulous')()(PLUGIN_NAME + ':utils');
+
+var sourceMapUrlRegEx = /\/\/\# sourceMappingURL\=.*/g
+
+
+var getCommentFormatter = function (file) {
+  var extension = file.relative.split('.').pop(),
+    fileContents =  file.contents.toString(),
+    newline =  detectNewline.graceful(fileContents || ''),
+    commentFormatter = function(url) {
+      return '';
+    };
+
+  if (file.sourceMap.preExisting){
+    debug('preExisting commentFormatter');
+    commentFormatter = function(url) {
+      return file.sourceMap.preExisting;
+    };
+    return commentFormatter
+  }
+
+  switch (extension) {
+    case 'css':
+      debug('css commentFormatter');
+      commentFormatter = function(url) {
+        return newline + "/*# sourceMappingURL=" + url + " */" + newline;
+      };
+      break;
+    case 'js':
+      debug('js commentFormatter');
+      commentFormatter = function(url) {
+        return newline + "//# sourceMappingURL=" + url + newline;
+      };
+      break;
+    default:
+      debug('unknown commentFormatter')
+  }
+
+  return commentFormatter;
+}
+
+var getPreExisting = function(fileContent){
+  if(sourceMapUrlRegEx.test(fileContent)){
```

Comment:
I'm not sure why the tests fail in 0.10 and not in 0.12 and higher, but that behavior is not a bug, it is part of the JS RegExp API by design.  You can test out the example from that stackoverflow question in the CLI to see.  (I tried on v4.3.1 and the behavior is still there.)